### PR TITLE
fix(agent): Fix agent.secure.enabled flag

### PR DESCRIPTION
**Pull request by aroberts87**
## What this PR does / why we need it:
Setting `agent.secure.enabled=false` was not explicitly disabling the secure features in the generated agent configuration. Instead, the behavior was left up to the agent process's defaults, or a backend config push. This behavior is not desirable as a user who explicitly set `agent.secure.enabled=false` could have ended up with secure components running anyway. This fix ensures that setting `agent.secure.enabled=false` explicitly generates the configuration entries to force all secure components to be disabled.

## Checklist

[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]

- [x] Title of the PR starts with type and scope, (e.g. feat(agent,node-analyzer,sysdig-deploy):)
- [x] Chart Version bumped for the respective charts
- [x] Check GithubAction checks (like lint) to avoid merge-check stoppers
- [x] All test files are added in the tests folder of their respective chart and have a "_test" suffix

Check Contribution guidelines in README.md for more insight.

### CODE REVIEW

File: charts/agent/templates/configmap.yaml [hunk 1]
```diff
@@ -13,20 +13,10 @@ data:
...
-  {{- end }}
-  {{- if .Values.secure.enabled }}
-      enabled: true
-    commandlines_capture:
-      enabled: true
-    memdump:
-      enabled: true
-  {{- end }}
+{{- if and (not .Values.secure.enabled) .Values.auditLog.enabled }}
+  {{- fail "auditLog.enabled cannot be true when secure.enabled is false" }}
```
Comment (lilx1ao) [inline charts/agent/templates/configmap.yaml]: tiny suggestion: might make sense to do it from within the agent.secureFeatures?

Comment (aroberts87) [reply charts/agent/templates/configmap.yaml]: Sure, I can do this. I think I tried that before and ended up with weird blank lines that I couldn't solve with adding `-`'s, but I agree with you it would look cleaner in the template. Let me try again. 💪🏻

Comment (lilx1ao) [reply charts/agent/templates/configmap.yaml]: 💪

Comment (aroberts87) [reply charts/agent/templates/configmap.yaml]: Done!

**Review by lilx1ao — APPROVED** [review]

File: charts/agent/templates/_helpers.tpl [hunk 2]
```diff
@@ -318,3 +318,31 @@ Use global sysdig tags for agent
         {{- end -}}
     {{- end -}}
 {{- end -}}
+
+{{/*
+Determine Sysdig Secure features that need to be enabled/disabled
+*/}}
+{{- define "agent.secureFeatures" }}
+    {{- $secureBlockConfig := dict "security" (dict
+        "enabled" .Values.secure.enabled
```
Comment (balinaveen) [inline charts/agent/templates/_helpers.tpl:337]: May want to explain why secure.enabled = true and false are not mirror images.
For secure.enabled, only security.enabled is set to true (along with the k8s_audit_server_enabled set to whatever the auditLog.enabled is)
For secure.enabled=false, all known secure features are disabled.

The explanation is that for secure.enabled=true we depend on the backend to enable the appropriate secure features for the customer. And since we don't know a priori what secure features may be enabled when secure.enabled=true we disable them all when secure.enabled=false

Comment (aroberts87) [reply charts/agent/templates/_helpers.tpl:337]: Done!

Comment (balinaveen) [reply charts/agent/templates/_helpers.tpl:337]: Thanks
